### PR TITLE
a new branch to enable the user to store data temporary

### PR DESCRIPTION
**Pull request by ogomb**
Please review the changes 

I have included a test file to test functional and unit tests.
I have also changed the app/views.py to correctly route and render the web pages in the template folder.

the user will be able to signup to the system first then be redirected to login page, the user data will be stored in a set since it allows for unique values, the login details will be compared with values in the set.

user will be able to add items to bucket list and also delete items from them ...(the back-end implementation for this is a python list)

### CODE REVIEW

File: app/views.py [hunk 1]
```diff
@@ -1,22 +1,54 @@
-from flask import render_template, request
+from flask import render_template, request, redirect, url_for
 
 from app import app
 
+users = set()
```
Comment (amutava) [inline app/views.py:5]: Why did you decide to use sets for users?

File: app/views.py [hunk 2]
```diff
@@ -1,22 +1,54 @@
...
+    if request.method == 'POST':
+        username = str(request.form.get('inputName'))
+        email = str(request.form.get('inputEmail'))
+        password = str(request.form.get('inputPassword'))
+
+        if username:
+            users.add(username)
+            users.add(email)
+            users.add(password)
+            print users
```
Comment (amutava) [inline app/views.py:25]: Are you checking the edge case where a user doesn't populate email and password in their form. Adding all the details the user provides in a set makes an implication that two people cannot have the same bucketlist, I would recommend using objects.

File: app/views.py [hunk 3]
```diff
@@ -1,22 +1,54 @@
...
+            return redirect(url_for('login'))
     return render_template('signup.html')
 
+@app.route('/login', methods=['GET', 'POST'])
+def login():
+    if request.method == 'POST':
+        email = str(request.form.get('inputEmail'))
+        password = str(request.form.get('inputPassword'))
+        if email in users and password in users:
```
Comment (amutava) [inline app/views.py:34]: Storing data in the set still implies two people cannot have the same password. Kindly review the appropriate use of data stuctures. I would recommend you check the use of objects.

File: tests/unit_test.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+    
+def test_root_url_has_correct_html():
+    client = app.test_client()
+    rsp = client.get('/')
+    assert rsp.status == '200 OK'
+    html = rsp.get_data(as_text=True)
+    assert'<title>BucketList</title>', html
+
+def test_sign_up_page_accepts_post_request():
```
Comment (amutava) [inline tests/unit_test.py:11]: For your tests check the use of assertEqual, assertIn

**Comment by tonykurya** [comment]
I like how the app looks. Great work there Ogomb.

**Comment by ghost** [comment]
Good work

**Pull request merged**